api-pre2.2/pom.xml

```diff
@@ -14,6 +14,43 @@
 	<name>Core Apps Module API-pre2.2</name>
 	<description>API project for CoreApps</description>
 
+	<dependencies>
+
+		<!-- Begin OpenMRS core -->
+
+		<dependency>
+			<groupId>org.openmrs.web</groupId>
+			<artifactId>openmrs-web</artifactId>
+			<type>jar</type>
+		</dependency>
+
+		<!-- End OpenMRS core -->
+
+		<!-- Begin OpenMRS modules -->
+
```

Comment:
Are all these dependencies needed?

Reply:
@dkayiwa I believe they are. the build fails if i exclude them.

Reply:
Are you sure for each of them?

Reply:
Yes, i am.

Reply:
Remove just one and compile. If u get errors, put it back, then go to the next and do the same.

Reply:
Which ones seem unnecessary so I can share compile log when running without them?

Reply:
Just do what i told you before.

Reply:
@dkayiwa alright. I have gone through each removing and putting back. They are all causing the build to fail.

Reply:
Did you remove all of them at once? Or just one at a time?
Can you list the ones you removed and failed?
And by the way, by removing one at a time, i mean when all the rest are back in.

Reply:
Oh sorry about that. You are correct.
I was not doing a full build

Reply:
🤣. Here's is what I did comment out one dependency, run test and when if it fails. uncomment it and go to the next dependency. do the same cycle till the last one. All of them failed when removed.

Reply:
That's alright.

Reply:
@dkayiwa what's the direction from here onwards?

Reply:
Did you say that you are able to add new diagnosis via the user interface and have them successfully saved in the new encounter_diagnosis table?

Reply:
I have one entry saved. Let me just double check with a few more and get back to you.

Reply:
@dkayiwa It add to the encounter_diagnosis table but i notice that after saving the first time, the subsequent diagnoses don't have the auto-suggest feature while being added.

Reply:
Can you investigate that?

Reply:
Alright. I will give you a ping when i have have some results.